src/game/client/draw/w3dmodeldraw.cpp

```diff
@@ -1154,3 +1179,1913 @@ void W3DModelDrawModuleData::Parse_Condition_State(INI *ini, void *instance, voi
+        new_state->Validate_Stuff(
+            m_renderObject, Get_Drawable()->Get_Scale(), Get_W3D_Model_Draw_Module_Data()->m_extraPublicBone);
+        Rebuild_Weapon_RecoilInfo(new_state);
+        Do_Hide_Show_Sub_Objs(&new_state->m_hideShowVec);
+    }
+
+    Hide_All_Headlights(m_isDaytime);
+    const ModelConditionInfo *old_state = m_curState;
+    m_curState = new_state;
+    m_nextState = next;
+    m_loopDuration = -1;
+    Adjust_Animation(old_state, fraction);
+}
+
+void W3DModelDraw::Replace_Model_Condition_State(BitFlags<MODELCONDITION_COUNT> const &c)
+{
+    m_isDaytime = !c.Test(MODELCONDITION_NIGHT);
+    const ModelConditionInfo *info = Find_Best_Info(c);
+
+    if (info != nullptr) {
+        Set_Model_State(info);
+    }
+
+    Hide_All_Headlights(m_isDaytime);
+}
+
+void W3DModelDraw::Set_Selectable(bool selectable)
+{
+    if (m_renderObject != nullptr) {
+        if (selectable) {
+            m_renderObject->Set_Collision_Type(m_renderObject->Get_Collision_Type() | COLLISION_TYPE_1);
+        } else {
+            m_renderObject->Set_Collision_Type(m_renderObject->Get_Collision_Type() & ~COLLISION_TYPE_1);
+        }
+    }
+}
+
+void W3DModelDraw::Replace_Indicator_Color(int color)
+{
+    if (Get_W3D_Model_Draw_Module_Data()->m_okToChangeModelColor && Get_Render_Object()) {
+        int c;
+
+        if (color) {
+            c = color | 0xFF000000;
+        } else {
+            c = 0;
+        }
+
+        if (c != m_hexColor) {
+            m_hexColor = c;
+            const ModelConditionInfo *state = m_curState;
+            m_curState = nullptr;
+            m_nextState = nullptr;
+            m_loopDuration = -1;
+            Set_Model_State(state);
+        }
+    }
+}
+
+bool W3DModelDraw::Client_Only_Get_Render_Obj_Info(Coord3D *pos, float *radius, Matrix3D *transform) const
+{
+    if (m_renderObject == nullptr) {
+        return false;
+    }
+
+    Vector3 v = m_renderObject->Get_Position();
+    pos->x = v.X;
+    pos->y = v.Y;
+    pos->z = v.Z;
+    *transform = m_renderObject->Get_Transform();
+    *radius = m_renderObject->Get_Bounding_Sphere().Radius;
+    return true;
+}
+
+bool W3DModelDraw::Get_Projectile_Launch_Offset(BitFlags<MODELCONDITION_COUNT> const &c,
+    WeaponSlotType wslot,
+    int ammo_index,
+    Matrix3D *launch_pos,
+    WhichTurretType tur,
+    Coord3D *turret_rot_pos,
+    Coord3D *turret_pitch_pos) const
+{
+    const ModelConditionInfo *info = Find_Best_Info(c);
+
+    if (info) {
+        W3DModelDrawModuleData *data = Get_W3D_Model_Draw_Module_Data();
+        info->Validate_Stuff(nullptr, Get_Drawable()->Get_Scale(), data->m_extraPublicBone);
+        captainslog_dbgassert(info->m_transition == 0,
+            "It is never legal to getProjectileLaunchOffset from a Transition state (they vary on a per-client basis)... "
+            "however, we can fix this");
+        captainslog_dbgassert(ammo_index >= 0, "specificBarrelToUse should now always be explicit");

```

Comment:
What does "specificBarrelToUse" refer to? There is no trace of it in this code.

Reply:
fixed.